cmd/zpool/zpool_vdev.c

```diff
@@ -548,7 +548,7 @@ is_shorthand_path(const char *arg, char *path,
 			return (0);
 	}
 
-	strlcpy(path, arg, sizeof (path));
+	strlcpy(path, arg, path_size);
```

Comment:
Yeah, this is clearly wrong and we need to use a passed size.  Similarly you should pass the new `path_size` to `zfs_resolve_shortname()` above and remove the assumption that the buffer is MAXPATHLEN.